### PR TITLE
python3Packages.pytest-md-report: init at 0.3.0

### PR DESCRIPTION
**Pull request by rrbutani**
###### Description of changes

`pytest-md-report`: https://pypi.org/project/pytest-md-report/

> A pytest plugin to make a test results report with Markdown table format.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/pytest-md-report/default.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+buildPythonPackage rec {
+  pname = "pytest-md-report";
+  version = "0.3.0";
+  format = "setuptools";
+
+  disabled = pythonOlder "3.6";
+
+  src = fetchPypi {
+    pname = "pytest-md-report";
+    inherit version;
```
Comment (fabaff) [inline pkgs/development/python-modules/pytest-md-report/default.nix:22]:
```suggestion
    inherit pname version;
```

File: pkgs/development/python-modules/pytest-md-report/default.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    hash = "sha256-ra88WXG6+xHSjOcy6tdYjvpKpNlvu6lq+sZckLadAlU=";
+  };
+
+  propagatedBuildInputs = [
+    pytablewriter
+    pytest
```
Comment (fabaff) [inline pkgs/development/python-modules/pytest-md-report/default.nix:28]: Sometimes it's necessary that we are able to override pytest. Thus it should go to `buildInputs`

```suggestion

  buildInputs = [
    pytest
  ];

  propagatedBuildInputs = [
    pytablewriter
```

File: pkgs/development/python-modules/pytest-md-report/default.nix [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    tcolorpy
+    typepy
+
+    setuptools
```
Comment (fabaff) [inline pkgs/development/python-modules/pytest-md-report/default.nix:32]: At first glance I don't see that `pkg_resources` or alike is used in the code.

File: pkgs/development/python-modules/pytest-md-report/default.nix [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  ];
+
+  nativeCheckInputs = [ pytestCheckHook ];
+
+  pythonImportsCheck = [ "pytest_md_report" ];
+
+  meta = with lib; {
+    description = "A pytest plugin to make a test results report with Markdown table format";
+    homepage = "https://github.com/thombashi/pytest-md-report";
```
Comment (fabaff) [inline pkgs/development/python-modules/pytest-md-report/default.nix:41]:
```suggestion
    homepage = "https://github.com/thombashi/pytest-md-report";
    changelog = "https://github.com/thombashi/pytest-md-report/releases/tag/v0.3.0";
```

File: pkgs/development/python-modules/pytest-md-report/default.nix (hunk 3 above)
Comment (rrbutani) [reply pkgs/development/python-modules/pytest-md-report/default.nix:32]: good catch, thanks!

File: pkgs/development/python-modules/pytest-md-report/default.nix (hunk 2 above)
Comment (rrbutani) [reply pkgs/development/python-modules/pytest-md-report/default.nix]: ah, I didn't consider that; fixed in the latest push, thanks

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 218706` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.pytest-md-report</li>
    <li>python311Packages.pytest-md-report</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.maintainers` is set
- [x] package builds (see https://github.com/NixOS/nixpkgs/pull/218706#issuecomment-1454257712)
- [ ] all depending packages build
- [x] tested on x86_64

Looks good to me.

**Pull request merged**